plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java

```diff
@@ -1786,4 +1800,54 @@ public void testIncorrectIcebergFileSizes()
 
         dropTable("test_iceberg_file_size");
     }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        throw new SkipException("Iceberg connector does not support column default values");
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getTrinoTypeName();
+        if (typeName.equals("tinyint")
+                || typeName.equals("smallint")
+                || typeName.startsWith("char(")) {
+            // These types are not supported by Iceberg
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+
+        if (typeName.startsWith("decimal(") || typeName.equals("time")) {
+            // Cases covered by testDecimal method
+            return Optional.empty();
+        }
+
+        if (typeName.equals("timestamp")) {
+            return Optional.of(new DataMappingTestSetup("timestamp(6)", "TIMESTAMP '2020-02-12 15:03:00'", "TIMESTAMP '2199-12-31 23:59:59.999999'"));
```

Comment:
Please add a comment about why this is required. Same for the `timestamp(3) with time zone` branch.